src/components/SliderField/SliderField.tsx

```diff
@@ -0,0 +1,204 @@
+  range?: boolean
+  onChange?: (
+    ev: React.ChangeEvent<HTMLInputElement>,
+    data: number | [number, number]
+  ) => void
+  onMouseUp?: (
+    ev: React.MouseEvent<HTMLInputElement>,
+    data: number | [number, number]
+  ) => void
+}
+
+export enum SliderLastInteraction {
+  'from',
+  'to'
+}
+export enum SliderDefault {
+  MIN = 0,
+  MAX = 100,
+  FROM = 0,
+  TO = 100
+}
+
+export type SliderFieldState = {
+  from: number
+  to: number
+  lastInteraction: SliderLastInteraction
+}
+
+export class SliderField extends React.PureComponent<
+  SliderFieldProps,
+  SliderFieldState
+> {
+  state = {
+    from: SliderDefault.FROM,
+    to: SliderDefault.TO,
+    lastInteraction: SliderLastInteraction.from
+  }
+
+  componentDidMount(): void {
+    if (this.props.defaultValue === undefined) {
+      this.setState((prevState) => {
+        return {
+          ...prevState,
+          from: !isNaN(this.props.min) ? this.props.min : prevState.from,
+          to: !isNaN(this.props.max) ? this.props.max : prevState.to
+        }
+      })
+    } else {
+      if (Array.isArray(this.props.defaultValue)) {
+        this.setState({
+          from: this.props.defaultValue[0],
+          to: this.props.defaultValue[1]
+        })
+      } else {
+        this.setState({
+          to: Number(this.props.defaultValue)
+        })
+      }
+    }
+  }
+
+  render(): JSX.Element {
+    const { header, className, onChange, onMouseUp, range } = this.props
+
+    const min = this.props.min || SliderDefault.MIN
+    const max = this.props.max || SliderDefault.MAX
+
+    const classes = ['dcl', 'sliderfield']
+    if (className) {
+      classes.push(className)
+    }
+
+    const handleChangeFrom = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const input = e.target
+
+      this.setState(
+        (prevState) => {
+          return {
+            ...prevState,
+            from: Math.min(Number(input.value), prevState.to),
+            lastInteraction: SliderLastInteraction.from
+          }
+        },
+        () => {
+          if (onChange) {
+            onChange(e, [this.state.from, this.state.to])
+          }
+        }
+      )
+    }
+
+    const handleChangeTo = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const input = e.target
+
+      this.setState(
+        (prevState) => {
+          return {
+            ...prevState,
+            to: Math.max(Number(input.value), prevState.from),
+            lastInteraction: SliderLastInteraction.to
+          }
+        },
+        () => {
+          if (onChange) {
+            onChange(
+              e,
+              range ? [this.state.from, this.state.to] : this.state.to
+            )
+          }
+        }
+      )
+    }
+
+    const handleMouseUp = (e: React.MouseEvent<HTMLInputElement>) => {
+      if (onMouseUp) {
+        onMouseUp(e, range ? [this.state.from, this.state.to] : this.state.to)
+      }
+    }
```

Comment:
These 3 handle functions
- handleChangeFrom
- handleChangeTo
- handleMouseUp

are being re-defined upon each re-render, because we're not using hooks and `useCallback`s here. Maybe we can add them as a instance method and use them below like

```ts
export class SliderField extends React.PureComponent {
  handleChangeFrom = () => {
    (...)
  }

  render() {
    (...)
     onChange={this.handleChangeFrom}
```

what do you think?

Reply:
done!